MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -28,35 +28,60 @@ public void Start(object hint = null)
 
         public virtual void ResetStart()
         {
+            Reset();
             Interlocked.Exchange(ref startHasCommenced, 0);
         }
+
+        protected virtual void Reset()
+        { }
     }
 
     public class MvxAppStart<TViewModel> : MvxAppStart
         where TViewModel : IMvxViewModel
     {
         protected readonly IMvxNavigationService NavigationService;
-        
-        public MvxAppStart(IMvxNavigationService navigationService)
+        protected readonly IMvxApplication Application;
+
+        public MvxAppStart(IMvxNavigationService navigationService, IMvxApplication application)
         {
             NavigationService = navigationService;
+            Application = application;
         }
 
         protected override void Startup(object hint = null)
+        {
+            ApplicationStartup();
+
+            NavigateToFirstViewModel(hint);
+        }
+
+        protected virtual void ApplicationStartup(object hint = null)
+        {
+            MvxLog.Instance.Trace("AppStart: Application Startup - On UI thread");
+            Application.Startup(hint);
+        }
+
+        protected virtual void NavigateToFirstViewModel(object hint)
```

Comment:
Would it make sense to let this return the `StartupViewModel`, and call that from `Startup`?

Reply:
Not sure I understand what you're thinking here - take a look at my refactor and let me know